catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddBrickFragment.kt

```diff
@@ -0,0 +1,154 @@
+            else -> CategoryBricksFactory()
+        }
+        val brickList = categoryBricksFactory.getBricks(selectedCategory, backgroundSprite == sprite, context)
+        adapter = PrototypeBrickAdapter(brickList)
+        listAdapter = adapter
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, menuInflater: MenuInflater) {
+        menu.findItem(R.id.comment_in_out).isVisible = false
+        super.onCreateOptionsMenu(menu, menuInflater)
+    }
+
+    override fun onDestroy() {
+        val actionBar = (activity as? AppCompatActivity)?.supportActionBar
+        val isRestoringPreviouslyDestroyedActivity = actionBar == null
+        if (!isRestoringPreviouslyDestroyedActivity) {
+            actionBar?.title = previousActionBarTitle
+        }
+        super.onDestroy()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        setupSelectedBrickCategory()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        if (listIndexToFocus != -1) {
+            listView.setSelection(listIndexToFocus)
+            listIndexToFocus = -1
+        }
+        listView.onItemClickListener = AdapterView.OnItemClickListener { parent: AdapterView<*>?, view: View?, position: Int, id: Long -> adapter?.getItem(position)?.let { addBrickToScript(it) } }
+    }
+
+    fun addBrickToScript(brick: Brick) {
+        if (projectManager.currentProject.isCastProject && CastManager.unsupportedBricks.contains(brick.javaClass)
+        ) {
+            ToastUtil.showError(activity, R.string.error_unsupported_bricks_chromecast)
+            return
+        }
+        try {
+            val brickToAdd = brick.clone()
+            addBrickListener?.addBrick(brickToAdd)
+            val fragmentTransaction = parentFragmentManager.beginTransaction()
+            val categoryFragment = parentFragmentManager.findFragmentByTag(BrickCategoryFragment.BRICK_CATEGORY_FRAGMENT_TAG)
+            if (categoryFragment != null) {
+                fragmentTransaction.remove(categoryFragment)
+                parentFragmentManager.popBackStack()
+            }
+            val addBrickFragment = parentFragmentManager.findFragmentByTag(ADD_BRICK_FRAGMENT_TAG)
+            if (addBrickFragment != null) {
+                fragmentTransaction.remove(addBrickFragment)
+                parentFragmentManager.popBackStack()
+            }
+            fragmentTransaction.commit()
+        } catch (e: CloneNotSupportedException) {
+            Log.e(tag, e.localizedMessage)
+            ToastUtil.showError(activity, R.string.error_adding_brick)
+        }
+    }
+
+    interface OnAddBrickListener {
+        fun addBrick(brick: Brick?)
+    }
+
+    companion object {
+        @JvmField
+        val ADD_BRICK_FRAGMENT_TAG = AddBrickFragment::class.java.simpleName
```

Comment:
Can this be private? If not, why is this used outside of the class? Please check :)